include/glow/Base/Type.h

```diff
@@ -275,6 +275,7 @@ enum class ElemKind : unsigned char {
   Int64ITy,      // 64-bit index type (int64_t)
   UInt8FusedQTy, // 8-bit quantized type with fused scale/offset (uint8_t)
   BoolTy,        // Bool type (bool)
+  AddrTy,        // Address type (uintptr_t)
```

Comment:
Host-side `uintptr_t` may be not enough to hold the address of the device buffer. I'd suggest using the existing `Int64ITy` for it or define `AddrTy` as int64_t.